src/domain/enrolment/__tests__/CreateEnrolmentPage.test.tsx

```diff
@@ -398,66 +398,74 @@ test('render and focuses error notification correctly', async () => {
   expect(screen.getByRole('textbox', { name: 'Nimi' })).toHaveFocus();
 });
 
-test('amount of adults is limited to 1 when there is 1 seat left after children', async () => {
-  render(<CreateEnrolmentPage />, {
-    mocks: mock3,
-    query: { eventId: eventId, occurrences: occurrenceIds },
-  });
-  await waitFor(() => {
-    expect(screen.getByLabelText(/lapsia/i)).toBeInTheDocument();
-  });
-  userEvent.type(screen.getByLabelText(/lapsia/i), '19');
-  userEvent.type(screen.getByLabelText(/aikuisia/i), '2');
-  userEvent.tab();
-  await waitFor(() => {
-    expect(
-      screen.getByText(
-        /Arvon tulee olla enintään 1 yhdessä lasten lukumäärän kanssa/i
-      )
-    ).toBeInTheDocument();
+describe('max group size validation of the Children and Adults -fields', () => {
+  const createEnrolmentForm = async (
+    childrenCount: string,
+    adultsCount: string
+  ) => {
+    render(<CreateEnrolmentPage />, {
+      mocks: mock3,
+      query: { eventId: eventId, occurrences: occurrenceIds },
+    });
+    await waitFor(() => {
+      expect(screen.getByLabelText(/lapsia/i)).toBeInTheDocument();
+    });
+    userEvent.type(screen.getByLabelText(/lapsia/i), childrenCount);
+    userEvent.type(screen.getByLabelText(/aikuisia/i), adultsCount);
+    userEvent.tab();
+  };
+
+  test('both of the fields are greater than the max group size', async () => {
+    createEnrolmentForm('21', '22');
+    await waitFor(() => {
+      expect(
+        screen.getAllByText(
+          /Lasten ja aikuisten yhteislukumäärän tulee olla enintään 20/i
+        )
+      ).toHaveLength(2);
+    });
   });
-});
 
-test('amount of adults is limited to 0 when children fills the group in whole', async () => {
-  render(<CreateEnrolmentPage />, {
-    mocks: mock3,
-    query: { eventId: eventId, occurrences: occurrenceIds },
-  });
-  await waitFor(() => {
-    expect(screen.getByLabelText(/lapsia/i)).toBeInTheDocument();
+  test('one field is greater than the max group size and another one is (still) empty', async () => {
+    createEnrolmentForm('21', '');
+    await waitFor(() => {
+      expect(
+        screen.getByText(
+          /Lasten ja aikuisten yhteislukumäärän tulee olla enintään 20/i
+        )
+      ).toBeInTheDocument();
+    });
+    expect(screen.getByText(/Tämä kenttä on pakollinen/i)).toBeInTheDocument();
   });
-  userEvent.type(screen.getByLabelText(/lapsia/i), '20');
-  userEvent.type(screen.getByLabelText(/aikuisia/i), '2');
-  userEvent.tab();
-  await waitFor(() => {
+
+  test('both the fields are valid as a single, but the total is greater than the maximum group size', async () => {
+    createEnrolmentForm('19', '18');
```

Comment:
One more minor improvement would be to pass one object instead of 2 parameters. Here it is not very evident what these parameters mean and you have to check the function declaration and remember (or hovering in vscode)

```
createEnrolment({ childrenCount: '19', adultCount: '18' })
```

just a small suggestion, no need to this change if you disagree :)

Reply:
It would also make a lot of repeat 😉. The createEnrolment is declared inside the describe-function of max group size validation, so I think I will leave it. Because the values are summed, it doesn't even make much difference in which order them are used.

Reply:
fair enough :)